esphome/components/remote_base/dyson_protocol.cpp

```diff
@@ -0,0 +1,71 @@
+#include "dyson_protocol.h"
+#include "esphome/core/log.h"
+
+#include <cinttypes>
+
+namespace esphome {
+namespace remote_base {
+
+static const char *const TAG = "remote.dyson";
+
+// pulsewidth [µs]
+constexpr uint32_t PW_MARK_US = 780;
+constexpr uint32_t PW_SHORT_US = 720;
+constexpr uint32_t PW_LONG_US = 1500;
+constexpr uint32_t PW_START_US = 2280;
+
+// MSB of 15 bit dyson code
+constexpr uint16_t MSB_DYSON = (1 << 14);
+
+// required symbols in transmit buffer = (start_symbol + 15 data_symbols)
+constexpr int32_t N_SYMBOLS_REQ = 2u * (1 + 15);
```

Comment:
Should keep this unsigned
```suggestion
constexpr uint32_t N_SYMBOLS_REQ = 2u * (1 + 15);
```